File: api/models/customer.go
package models

import "gorm.io/gorm"

type Address struct {
	Street     string
	StreetNr   string
	PostalCode string
	City       string
	Country    string
}

type Customer struct {
	gorm.Model
	Title *string
	// FormOfAddress *string TODO
	FirstName string
	LastName  string
	Company   *string
	Email     *string
	Phone     *string
	Address   *Address  `gorm:"embedded"`
	Invoices  []Invoice `json:",omitempty"`

	MicrosoftContactId *string
}

File: docx-renderer/models/customer.py
from pydantic import BaseModel


class Address(BaseModel):
    street: str
    streetNr: str
    postalCode: str
    city: str
    country: str


class Customer(BaseModel):
    title: str | None = None
    firstName: str
    lastName: str
    company: str
    address: Address
